src/lib/components/search.svelte

```diff
@@ -11,27 +11,20 @@
   export let id = `${label.toLocaleUpperCase()}-input`;
 </script>
 
-<form
-  on:submit|preventDefault
-  class="relative flex items-center rounded-lg border-2 bg-white transition-colors focus-within:border-blue-700"
-  role="search"
->
-  {#if icon}
-    <Icon name="search" class="ml-4 flex items-center text-gray-700" />
-  {/if}
-  <label for={id} class="hidden">{label}</label>
-  <input
+<form on:submit|preventDefault role="search" class="flex items-center">
+  <Input
+    class="w-full"
+    unroundRight
+    {label}
     {id}
+    labelHidden
+    icon={icon ? 'search' : null}
```

Comment:
This ternary feels weird.

Reply:
`icon` is a boolean prop that controls whether the magnifying glass icon displays in the input. `icon={icon && 'search'}` won't do, because then `false` would be passed to `icon` on `<Input />` which is not valid. Could change `null` to `undefined` or `''` but other than that I'm not sure how to make this better.